starknet_py/utils/typed_data.py

```diff
@@ -360,6 +377,62 @@ def encode_bool(value: Union[bool, str, int]) -> int:
     raise ValueError(f"Expected boolean value, got [{value}].")
 
 
+def is_digit_string(s: str, signed=False) -> bool:
+    if signed:
+        return bool(re.fullmatch(r"-?\d+", s))
+    return bool(re.fullmatch(r"\d+", s))
+
+
+def encode_u128(value: Union[str, int]) -> int:
+    def is_in_range(n: int):
+        return 0 <= n < 2**128
+
+    if isinstance(value, int):
+        if is_in_range(value):
+            return value
+        raise ValueError(f"Value [{value}] is out of range for '{BasicType.U128}'.")
+
+    if isinstance(value, str):
+        int_value = None
+
+        if value.startswith("0x"):
+            int_value = int(value, 16)
+        elif is_digit_string(value):
+            int_value = int(value)
+
+        if int_value is not None and is_in_range(int_value):
+            return int_value
+
+    raise ValueError(f"Value [{value}] is out of range for '{BasicType.U128}'.")
```

Comment:
This is the sole error we return. Therefore, any incorrect string value not starting with `0x` will result in `out of range` error. An example of this would be the `example` case in `test_encode_invalid_u128`. A better approach would be to handle the invalid number error separately. The same for `encode_i128`.